charts/localstack/templates/ingress.yaml

```diff
@@ -26,15 +26,17 @@ spec:
   tls:
   {{- range .Values.ingress.tls }}
     - hosts:
+      {{- $base := . }}
       {{- range .hosts }}
-        - {{ . | quote }}
+        - {{ tpl . $base | quote }}
```

Comment:
You don't need to preserve the original context, you can refer to it with `$`:
```patch
diff --git a/charts/localstack/templates/ingress.yaml b/charts/localstack/templates/ingress.yaml
index 285c29b..da46633 100644
--- a/charts/localstack/templates/ingress.yaml
+++ b/charts/localstack/templates/ingress.yaml
@@ -26,17 +26,15 @@ spec:
   tls:
   {{- range .Values.ingress.tls }}
     - hosts:
-      {{- $base := . }}
       {{- range .hosts }}
-        - {{ tpl . $base | quote }}
+        - {{ tpl . $ | quote }}
       {{- end }}
       secretName: {{ .secretName }}
   {{- end }}
 {{- end }}
   rules:
-  {{- $base := . }}
   {{- range .Values.ingress.hosts }}
-    - host: {{ tpl .host $base | quote }}
+    - host: {{ tpl .host $ | quote }}
       http:
         paths:
           {{- range .paths }}
```

Reply:
That's good to know! 
Fixed